src/game/game_gui.cpp

```diff
@@ -0,0 +1,467 @@
+				}
+				break;
+			}
+		}
+	}
+
+	void OnPaint() override
+	{
+		if (this->closing_dropdown) {
+			this->closing_dropdown = false;
+			this->clicked_dropdown = false;
+		}
+		this->DrawWidgets();
+	}
+
+	void OnClick(Point pt, int widget, int click_count) override
+	{
+		if (widget >= WID_GSC_TEXTFILE && widget < WID_GSC_TEXTFILE + TFT_END) {
+			if (GameConfig::GetConfig() == nullptr) return;
+
+			ShowScriptTextfileWindow((TextfileType)(widget - WID_GSC_TEXTFILE), (CompanyID)OWNER_DEITY);
+			return;
+		}
+
+		switch (widget) {
+			case WID_GSC_GSLIST: {
+				this->InvalidateData();
+				if (click_count > 1 && _game_mode != GM_NORMAL) ShowAIListWindow((CompanyID)OWNER_DEITY);
+				break;
+			}
+
+			case WID_GSC_CHANGE:  // choose other Game Script
+				ShowAIListWindow((CompanyID)OWNER_DEITY);
+				break;
+
+			case WID_GSC_CONTENT_DOWNLOAD:
+				if (!_network_available) {
+					ShowErrorMessage(STR_NETWORK_ERROR_NOTAVAILABLE, INVALID_STRING_ID, WL_ERROR);
+				} else {
+					ShowNetworkContentListWindow(nullptr, CONTENT_TYPE_GAME);
+				}
+				break;
+			case WID_GSC_SETTINGS: {
+				const NWidgetBase* wid = this->GetWidget<NWidgetBase>(WID_GSC_SETTINGS);
+				int num = (pt.y - wid->pos_y) / this->line_height + this->vscroll->GetPosition();
+				if (num >= (int)this->visible_settings.size()) break;
+
+				VisibleSettingsList::const_iterator it = this->visible_settings.begin();
+				for (int i = 0; i < num; i++) it++;
+				const ScriptConfigItem config_item = **it;
+				if (!this->IsEditableItem(config_item)) return;
+
+				if (this->clicked_row != num) {
+					this->CloseChildWindows(WC_QUERY_STRING);
+					HideDropDownMenu(this);
+					this->clicked_row = num;
+					this->clicked_dropdown = false;
+				}
+
+				bool bool_item = (config_item.flags & SCRIPTCONFIG_BOOLEAN) != 0;
+
+				int x = pt.x - wid->pos_x;
+				if (_current_text_dir == TD_RTL) x = wid->current_x - 1 - x;
+				x -= 4;
+
+				/* One of the arrows is clicked (or green/red rect in case of bool value) */
+				int old_val = this->gs_config->GetSetting(config_item.name);
+				if (!bool_item && IsInsideMM(x, 0, SETTING_BUTTON_WIDTH) && config_item.complete_labels) {
+					if (this->clicked_dropdown) {
+						/* unclick the dropdown */
+						HideDropDownMenu(this);
+						this->clicked_dropdown = false;
+						this->closing_dropdown = false;
+					} else {
+						const NWidgetBase* wid = this->GetWidget<NWidgetBase>(WID_GSC_SETTINGS);
+						int rel_y = (pt.y - (int)wid->pos_y) % this->line_height;
+
+						Rect wi_rect;
+						wi_rect.left = pt.x - (_current_text_dir == TD_RTL ? SETTING_BUTTON_WIDTH - 1 - x : x);
+						wi_rect.right = wi_rect.left + SETTING_BUTTON_WIDTH - 1;
+						wi_rect.top = pt.y - rel_y + (this->line_height - SETTING_BUTTON_HEIGHT) / 2;
+						wi_rect.bottom = wi_rect.top + SETTING_BUTTON_HEIGHT - 1;
+
+						/* For dropdowns we also have to check the y position thoroughly, the mouse may not above the just opening dropdown */
```

Comment:
This comment doesn't make sense. Words out of order?

Reply:
Same reasoning as https://github.com/OpenTTD/OpenTTD/pull/10058/#discussion_r1015415356. Was copied over from `ai_gui.cpp`

Reply:
It should either be changed to make sense or removed, then. No sense keeping it the way it is. 🙂